OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -127,4 +127,26 @@ void ModelComponent::updateFromXMLNode(SimTK::Xml::Element& aNode,
     Super::updateFromXMLNode(aNode, versionNumber);
 }
 
+// Base class implementations of virtual methods for scaling.
+void ModelComponent::preScale(const SimTK::State& s, const ScaleSet& scaleSet)
+{   extendPreScale(s, scaleSet); }
+
+void ModelComponent::scale(const SimTK::State& s, const ScaleSet& scaleSet)
+{   extendScale(s, scaleSet); }
+
+void ModelComponent::postScale(const SimTK::State& s, const ScaleSet& scaleSet)
+{   extendPostScale(s, scaleSet); }
+
+const SimTK::Vec3 ModelComponent::
```

Comment:
You can remove the `const` since you are returning a copy anyways. Otherwise to avoid copies, you could return a const reference. You could define  `const static SimTK::Vec3 InvalidScaleFactors{SimTK::NaN}`  as a private data member to return in the not found case (e.g.`return InvalidScaleFactors;`).  This is just a minor fine-tuning.

Reply:
I removed the `const` and added a note (somewhat obvious) above the `return` statement. I think defining `InvalidScaleFactors` and returning a const reference is unnecessary for what's going on here. (I would also need to change the calling methods to use something like `areScaleFactorsValid()` because checking `scaleFactors.isNaN()` would make an assumption about the default return value that, presumably, could change. If we're concerned about performance, the next thing to improve is the string comparison part in `ModelComponent::getScaleFactors()`.